nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+    Returns
+    -------
+    labels : array of shape (n_voxels,),
+        A map of values on voxels used to identify the corresponding model.
+
+    results : dict,
+        Keys correspond to the different labels values
+        values are RegressionResults instances corresponding to the voxels.
+    """
+
+    if Y.shape[0] != X.shape[0]:
+        raise ValueError(
+            'The number of rows of Y should match the number of rows of X.'
+            ' You provided X with shape {0} and Y with shape {1}'.\
+                format(X.shape, Y.shape))
+
+    # Create the model
+    ols_result = OLSModel(X).fit(Y)
+    labels = np.zeros(Y.shape[1])
+    results = {0.0: ols_result}
+
+    return labels, results
+
+
+class SecondLevelModel(BaseEstimator, TransformerMixin, CacheMixin):
+    """ Implementation of the General Linear Model for multiple subject
+    fMRI data
+
+    Parameters
+    ----------
+
+    mask: Niimg-like, NiftiMasker or MultiNiftiMasker object, optional,
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters. Mask automatic computation assumes first level imgs have
+        already been masked.
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    memory: string, optional
+        Path to the directory used to cache the masking process and the glm
+        fit. By default, no caching is done. Creates instance of joblib.Memory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    verbose : integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    n_jobs : integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    """
+    def __init__(self, mask=None, smoothing_fwhm=None,
+                 memory=None, memory_level=1, verbose=1,
+                 n_jobs=1, minimize_memory=True):
+        self.mask = mask
+        self.smoothing_fwhm = smoothing_fwhm
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.n_jobs = n_jobs
+        self.minimize_memory = minimize_memory
+        self.labels_ = None
+        self.results_ = None
+
+    def fit(self, first_level_input, first_level_conditions=None,
+            confounds=None, design_matrix=None,
+            first_level_conditions_name=None):
+        """ Fit the GLM
+
+        1. create design matrix
+        2. do a masker job: fMRI_data -> Y
+        3. fit regression to (Y, X)
+
+        Parameters
+        ----------
+        first_level_input: list of `FirstLevelModel` objects or pandas
```

Comment:
as discussed in the example, I am afraid that this terminology will confuse users, as it can be interpreted "input to first level analysis". `second_level_input` makes more sense to me. Do you agree ?